lib/dynflow/execution_plan.rb

```diff
@@ -199,16 +199,17 @@ def sub_plans_count
       persistence.find_execution_plan_counts(filters: { 'caller_execution_plan_id' => self.id })
     end
 
-    def rescue_plan_id
+    def prepare_for_rescue
       case rescue_strategy
       when Action::Rescue::Pause
-        nil
+        return :paused
       when Action::Rescue::Fail
-        update_state :stopped
-        nil
+        return :stopped
       when Action::Rescue::Skip
         failed_steps.each { |step| self.skip(step) }
-        self.id
+        return :running
+      else
+        return :paused
```

Comment:
None of these returns is really needed, no need to change, just saying.

Reply:
updated